### PR TITLE
Dijkstra & A*

### PR DESCRIPTION
**Pull request by 32monkeys**

### CODE REVIEW

File: .idea/deployment.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="PublishConfigData">
+    <serverData>
+      <paths name="wits">
```
Comment (lmbek) [inline .idea/deployment.xml:5]: why is this commited

File: .idea/misc.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <component name="ProjectKey">
     <option name="state" value="project://e2804f05-5315-4fc6-a121-c522a6c26470" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_13" default="true" project-jdk-name="13" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_12" default="false" project-jdk-name="13" project-jdk-type="JavaSDK">
```
Comment (lmbek) [inline .idea/misc.xml:6]: back to 13

File: src/files/models/AStar.java [hunk 3]
```diff
@@ -1,4 +1,115 @@
...
+        aStar.startAStar();
+    }
+
+    public void startAStar() {
+        // Create graph
+        //graph = this.makeDijkstraGraph();
+        graph = this.makeSmallGraphA();
+        // A
+        Vertex startNode = graph.getVertex("A");
+        Vertex endNode = graph.getVertex("F");
```
Comment (lmbek) [inline src/files/models/AStar.java:26]: Great but why is this F in test (we dont have F in that graph)

File: src/files/models/AStar.java [hunk 4]
```diff
@@ -1,4 +1,115 @@
...
+        myGraph.newEdge(C,E, 1,  3);
+        myGraph.newEdge(D,E, 6,  3);
+        myGraph.newEdge(E,D, 4,  3);
+
+        return myGraph;
+    }
+
+    public ArrayList<Vertex> aStar(Vertex startNode, Vertex endNode) {
+
+        TreeSet<Vertex> openTreeSet = new TreeSet<>(Comparator.comparingLong(Vertex::getF));
```
Comment (lmbek) [inline src/files/models/AStar.java:59]: is this to we can compare distances? What is getF

File: src/files/models/AStar.java [hunk 5]
```diff
@@ -1,4 +1,115 @@
...
+        ArrayList<Vertex> closedList = new ArrayList<>();
+
+        for (Vertex vertex : graph.getVertices()) {
+            vertex.setDistance(200);
```
Comment (lmbek) [inline src/files/models/AStar.java:63]: we can use infinity here. Will probably have to be reworked from 200 anyways, as our distance can be over 200 in our case.

File: src/files/models/AStar.java [hunk 6]
```diff
@@ -1,4 +1,115 @@
...
+            current = openTreeSet.first();
+            System.out.println(current.name + " " + current.distance);
+            openTreeSet.remove(current);
+
+            for (Edge edge : current.edges){
+
+                Vertex successor = edge.getToVertex();
+                int successorG = current.distance + successor.distance;
+                long successorH = heuristic(current,endNode);
+                long successorF = successorG + successorH;
```
Comment (lmbek) [inline src/files/models/AStar.java:82]: so F is from the formula. But maybe its good to add comments if the full word is not used

File: src/files/models/Dijkstra.java [hunk 7]
```diff
@@ -131,24 +94,34 @@ public Vertex dijkstra(Vertex startNode, Vertex endNode) {
...
                         + "  dist:  " + edge.getToVertex().distance);
             }
-            //graphTreeSet.forEach((vertex -> System.out.print(vertex.name + ": " + vertex.distance + "   |")));
+
             graphTreeSet.remove(current);
             System.out.println("removed:  " +current.name);
         }
 
-        return endNode;
+        Vertex resultCurrent = endNode;
```
Comment (lmbek) [inline src/files/models/Dijkstra.java:117]: not sure i understand what resultCurrent is used for

File: src/files/models/graph/Vertex.java [hunk 8]
```diff
@@ -20,4 +20,8 @@ public void addEdge(Edge edge){
     public void setDistance(int distance) { this.distance = distance; }
     public Vertex getPredecessor() { return predecessor; }
     public void setPredecessor(Vertex predecessor) { this.predecessor = predecessor; }
+
+    public long f;
```
Comment (lmbek) [inline src/files/models/graph/Vertex.java:24]: what is f, is it a function?

File: src/files/models/Dijkstra.java [hunk 9]
```diff
@@ -37,18 +37,23 @@ public void startDijkstra() {
         //Vertex startNode = graph.getVertex("J");
         //Vertex endNode = graph.getVertex("F");
 
-        Pair<Integer, Map<Vertex, Vertex>> results = dijkstra(startNode, endNode);
+        Vertex result = dijkstra(startNode, endNode);
```
Comment (lmbek) [inline src/files/models/Dijkstra.java]: Would be nice with something similar with the Pair. Like an arrayList of the Vertex's as we can get the distances or positions of x and y from the vertices in the arrayList.
(this is probably already done by you recently)

**Review by lmbek — APPROVED** [review]
Approved, i have added some feedback. I would like to get explained what F and G is in our formula as i'm just unsure of them myself.
Get the rest of the algorithm done, then we'll try to clean it together. Good job!

File: src/files/models/Dijkstra.java (hunk 9 above)
Comment (32monkeys) [reply src/files/models/Dijkstra.java]: wut?

File: src/files/models/Dijkstra.java (hunk 7 above)
Comment (32monkeys) [reply src/files/models/Dijkstra.java:117]: changed it so it returns an arraylist

File: src/files/models/AStar.java (hunk 5 above)
Comment (32monkeys) [reply src/files/models/AStar.java:63]: yea it was a quick test to make sure infinity wasn't causing problems again

File: .idea/misc.xml (hunk 2 above)
Comment (32monkeys) [reply .idea/misc.xml:6]: I don't have 13, so I will change it back to 12 anyways

File: .idea/deployment.xml (hunk 1 above)
Comment (32monkeys) [reply .idea/deployment.xml:5]: dunno, guess it isn't in the out folder

File: src/files/models/AStar.java (hunk 6 above)
Comment (32monkeys) [reply src/files/models/AStar.java:82]: There are no full words, this is maths my friend. f is function. g and h  just comes after f in the alphabet, that is all

**Pull request merged**